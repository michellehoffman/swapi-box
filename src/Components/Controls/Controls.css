@import url('https://fonts.googleapis.com/css?family=Kanit:800,900');

.controls {
  border: 1px solid rgba(255, 232, 31, 1);
  background-color: transparent;
  padding: 10px 20px;
  font-family: 'Kanit', sans-serif;
  font-size: 1.6rem;
  margin: 10px;
  color: rgba(255, 232, 31, 1);
  margin-bottom: 50px;
}

.active {
  background-color: rgba(255, 232, 31, 1);
  color: black;
}

#favorites-button {
  position: absolute;
  right: 20px;
  top: 20px;
}

#num-faves {
  color: black;
  background-color: rgba(255, 232, 31, 1);
  padding: 0 10px;
  margin-left: 5px
}

.active #num-faves {
  color: rgba(255, 232, 31, 1);
  background-color: black;
}

