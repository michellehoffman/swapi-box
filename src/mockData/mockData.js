export const mockData = {
  "people": {
    "count": 1, 
    "next": "https://swapi.co/api/people/?page=2", 
    "previous": null, 
    "results": [
      {
        "name": "Luke Skywalker", 
        "height": "172", 
        "mass": "77", 
        "hair_color": "blond", 
        "skin_color": "fair", 
        "eye_color": "blue", 
        "birth_year": "19BBY", 
        "gender": "male", 
        "homeworld": "https://swapi.co/api/planets/1/", 
        "films": [
          "https://swapi.co/api/films/2/", 
          "https://swapi.co/api/films/6/", 
          "https://swapi.co/api/films/3/", 
          "https://swapi.co/api/films/1/", 
          "https://swapi.co/api/films/7/"
        ], 
        "species": [
          "https://swapi.co/api/species/1/"
        ], 
        "vehicles": [
          "https://swapi.co/api/vehicles/14/", 
          "https://swapi.co/api/vehicles/30/"
        ], 
        "starships": [
          "https://swapi.co/api/starships/12/", 
          "https://swapi.co/api/starships/22/"
        ], 
        "created": "2014-12-09T13:50:51.644000Z", 
        "edited": "2014-12-20T21:17:56.891000Z", 
        "url": "https://swapi.co/api/people/1/"
      }
    ]
  },
  "planets": {
    "count": 1, 
    "next": "https://swapi.co/api/planets/?page=2", 
    "previous": null, 
    "results": [
      {
        "name": "Alderaan", 
        "rotation_period": "24", 
        "orbital_period": "364", 
        "diameter": "12500", 
        "climate": "temperate", 
        "gravity": "1 standard", 
        "terrain": "grasslands, mountains", 
        "surface_water": "40", 
        "population": "2000000000", 
        "residents": [
          "https://swapi.co/api/people/5/", 
          "https://swapi.co/api/people/68/", 
          "https://swapi.co/api/people/81/"
        ], 
        "films": [
          "https://swapi.co/api/films/6/", 
          "https://swapi.co/api/films/1/"
        ], 
        "created": "2014-12-10T11:35:48.479000Z", 
        "edited": "2014-12-20T20:58:18.420000Z", 
        "url": "https://swapi.co/api/planets/2/"
      }
    ]
  },
  "vehicles": {
    "count": 1, 
    "next": "https://swapi.co/api/vehicles/?page=2", 
    "previous": null, 
    "results": [
      {
        "name": "Sand Crawler", 
        "model": "Digger Crawler", 
        "manufacturer": "Corellia Mining Corporation", 
        "cost_in_credits": "150000", 
        "length": "36.8", 
        "max_atmosphering_speed": "30", 
        "crew": "46", 
        "passengers": "30", 
        "cargo_capacity": "50000", 
        "consumables": "2 months", 
        "vehicle_class": "wheeled", 
        "pilots": [], 
        "films": [
          "https://swapi.co/api/films/5/", 
          "https://swapi.co/api/films/1/"
        ], 
        "created": "2014-12-10T15:36:25.724000Z", 
        "edited": "2014-12-22T18:21:15.523587Z", 
        "url": "https://swapi.co/api/vehicles/4/"
      }
    ]
  }
}