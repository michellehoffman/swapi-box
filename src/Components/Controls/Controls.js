import React from 'react';
import './Controls.css';

const Controls = ({ favorites, dataToDisplay }) => {
  const getElement = e => {
    const element = e.target.innerText;
    dataToDisplay(element);
  }

  const num = favorites ? favorites.length : 0

  return (
    <div>
      <div>
        <button onClick={ getElement }>people</button>
        <button onClick={ getElement }>planets</button>
        <button onClick={ getElement }>vehicles</button>
      </div>
      <div>
        <button>favorites <span>{ num }</span></button>
      </div>
    </div> 
  )
}

export default Controls;