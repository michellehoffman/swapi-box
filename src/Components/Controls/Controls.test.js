import React from 'react';
import ReactDOM from 'react-dom';
import { shallow } from 'enzyme';
import mockData from '../../mockData/mockData';
import Controls from './Controls';

describe('Controls', () => {
  let wrapper;

  const mockFavorites = [{
    name: "LukeSkywalker",
    type: "people",
  }]

  const mockSetCurrent = jest.fn();

  beforeEach(() => {
    wrapper = shallow(<Controls favorites={ mockFavorites }
                                setCurrent={ mockSetCurrent }
                                active="people" />);
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
})