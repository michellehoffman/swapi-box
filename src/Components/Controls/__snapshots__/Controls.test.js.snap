// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Controls should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Controls />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div>
          <button
                    onClick={[Function]}
          >
                    people
          </button>
          <button
                    onClick={[Function]}
          >
                    planets
          </button>
          <button
                    onClick={[Function]}
          >
                    vehicles
          </button>
</div>,
        <div>
          <button>
                    favorites 
                    <span />
          </button>
</div>,
      ],
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <button
              onClick={[Function]}
>
              people
</button>,
            <button
              onClick={[Function]}
>
              planets
</button>,
            <button
              onClick={[Function]}
>
              vehicles
</button>,
          ],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "people",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "people",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "planets",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "planets",
            "type": "button",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "vehicles",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "vehicles",
            "type": "button",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <button>
            favorites 
            <span />
</button>,
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "favorites ",
              <span />,
            ],
          },
          "ref": null,
          "rendered": Array [
            "favorites ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": 0,
              },
              "ref": null,
              "rendered": 0,
              "type": "span",
            },
          ],
          "type": "button",
        },
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div>
            <button
                        onClick={[Function]}
            >
                        people
            </button>
            <button
                        onClick={[Function]}
            >
                        planets
            </button>
            <button
                        onClick={[Function]}
            >
                        vehicles
            </button>
</div>,
          <div>
            <button>
                        favorites 
                        <span />
            </button>
</div>,
        ],
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <button
                onClick={[Function]}
>
                people
</button>,
              <button
                onClick={[Function]}
>
                planets
</button>,
              <button
                onClick={[Function]}
>
                vehicles
</button>,
            ],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "people",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "people",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "planets",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "planets",
              "type": "button",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "vehicles",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "vehicles",
              "type": "button",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <button>
              favorites 
              <span />
</button>,
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "favorites ",
                <span />,
              ],
            },
            "ref": null,
            "rendered": Array [
              "favorites ",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": 0,
                },
                "ref": null,
                "rendered": 0,
                "type": "span",
              },
            ],
            "type": "button",
          },
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
